evalml/pipelines/time_series_regression_pipeline.py

```diff
@@ -95,6 +95,8 @@ def predict(self, X, y=None, objective=None):
         if self.estimator.predict_uses_y:
             y_arg = y
         predictions = self.estimator.predict(features_no_nan, y_arg).to_series()
+        predictions.index = y.index
```

Comment:
The polynomial detrender needs the predictions to have the same index as the data it was fit on.
I think this is another reason to do #1639

Reply:
I think unfortunately in this case `reindex_like` is not avaliable for series, but even if I'm wrong it'll fill NaNs in the places where the indices don't match. I think `reindex` does the same thing.